### ISSUE TITLE
Warn for insufficient `OPAQUE` annotations

### ISSUE THREAD

**Issue by DigitalBrains1** [clash-lang/clash-compiler#2653]
Currently we emit a warning when users forget to annotate their primitive with `OPAQUE` (or `NOINLINE` on older GHC's):
```
Bla2.hs:21:1: Warning: primitive Bla2.bla isn't marked OPAQUE.
```
We should emit warnings for all cases where people should add `OPAQUE`: anything annotated `Synthesize`, including `topEntity` and `testBench`; in fact, anything at all where Clash needs to find a name in the Core produced by GHC.

**Comment by martijnbastiaan** [comment]
Yes. I feel like this is a proper bug, so I've added the label!
